### PR TITLE
fix: Update recorded parent's HEAD after rebase

### PR DESCRIPTION
**Pull request by draftcode**

### CODE REVIEW

File: e2e_tests/helpers.go [hunk 1]
```diff
@@ -12,3 +14,11 @@ func RequireCurrentBranchName(t *testing.T, repo *git.Repo, name string) {
 	require.NoError(t, err, "failed to determine current branch name")
 	require.Equal(t, name, currentBranch, "expected current branch to be %q, got %q", name, currentBranch)
 }
+
+func GetStoredParentBranchState(t *testing.T, repo *git.Repo, name string) meta.BranchState {
```
Comment (travigd) [inline e2e_tests/helpers.go:18]: Not for this PR, but we should add a command to get this information that way we don't have to do this.

File: e2e_tests/helpers.go [hunk 2]
```diff
@@ -12,3 +14,11 @@ func RequireCurrentBranchName(t *testing.T, repo *git.Repo, name string) {
...
+	// We shouldn't do this as part of an E2E test, but it's hard to ensure otherwise.
+	db, err := jsonfiledb.OpenRepo(repo)
```
Comment (travigd) [inline e2e_tests/helpers.go:20]: Do we need to close this to close the file handle? (not sure if that's actually a part of the DB interface in hindsight... 🤔)

**Review by travigd — APPROVED** [review]
Thanks for finding/fixing this!!!!!!!

File: e2e_tests/helpers.go (hunk 2 above)
Comment (draftcode) [reply e2e_tests/helpers.go:20]: jsonfiledb doesn't have a file handle in it.

File: e2e_tests/helpers.go (hunk 1 above)
Comment (draftcode) [reply e2e_tests/helpers.go:18]: Sure

File: e2e_tests/helpers.go (hunk 2 above)
Comment (travigd) [reply e2e_tests/helpers.go:20]: Now that you say that I remember it lol. I wrote that code all of two weeks ago and remember none of it apparently.

**Pull request merged**